### PR TITLE
refactor(core): split mock.ts into __mocks__/*.ts

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

Since `utils/mock.ts` is too long, split it into multiple files.

## Linear Issue Reference

LOG-2063

## Testing

Pass all existing tests.

### CODE REVIEW

**Comment by wangsijie** [comment]
Better exclude this folder in codecov?

File: packages/core/jest.config.js [hunk 1]
```diff
@@ -12,6 +12,6 @@ module.exports = {
     '^@/(.*)$': '<rootDir>/src/$1',
     '^jose/(.*)$': '<rootDir>/node_modules/jose/dist/node/cjs/$1',
   },
-  coveragePathIgnorePatterns: ['/node_modules/', '/build/'],
+  coveragePathIgnorePatterns: ['/node_modules/', '/build/', '/src/__mocks__/'],
```
Comment (wangsijie) [inline packages/core/jest.config.js:15]: nit: can we move `__mocks__` to the root dir? @logto-io/eng

Comment (demonzoo) [reply packages/core/jest.config.js:15]: It's a best practice to put `__mocks__` folder immediately adjacent to the target module. E.g.

image

So I don't know, maybe we should even break it into several more `__mocks__` folders?

Comment (IceHe) [reply packages/core/jest.config.js:15]: @simeng-li any idea

Comment (IceHe) [reply packages/core/jest.config.js:15]: Conclusion: Keep it for now, and build excludes `src/__mocks__/`.

See Slack [thread](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1649389597380479)

Comment (IceHe) [reply packages/core/jest.config.js:15]: Updated. PTAL @wangsijie @demonzoo

**Review by wangsijie — APPROVED** [review]
LGTM

**Comment by IceHe** [comment]
Rebased master to resolve conflicts.

**Pull request merged**